app/controllers/PreferencesProxy.scala

```diff
@@ -0,0 +1,40 @@
+package controllers
+
+import com.gu.pandomainauth.PanDomainAuthSettingsRefresher
+import com.gu.workflow.util.{Code, Dev}
+import config.Config
+import play.api.Logging
+import play.api.libs.ws.WSClient
+import play.api.mvc.{BaseController, ControllerComponents}
+
+import scala.concurrent.ExecutionContext.Implicits.global
+
+/* Proxy for editorial-preferences, to help with latency since workflow-frontend sits behind CloudFront*/
+class PreferencesProxy(
+  override val config: Config,
+  override val controllerComponents: ControllerComponents,
+  override val wsClient: WSClient,
+  override val panDomainSettings: PanDomainAuthSettingsRefresher
+) extends BaseController with PanDomainAuthActions with Logging {
+
+  private def proxyRequest(relativePath: String) = APIAuthAction.async { request =>
+    val url = s"${config.preferencesUrl}/$relativePath"
+    wsClient.url(url)
+      .withHttpHeaders(request.headers.toSimpleMap.toSeq: _*)
```

Comment:
Will this pass through the original host header (for workflow) rather than the new host header (for preferences)?

If so, you might find that this works now but stops working when a seemingly innocuous change is made to how presence is served.

Reply:
great spot - I've made the `Host` header explicit now